addons/nuxeo-platform-pdf-utils/src/main/java/org/nuxeo/ecm/platform/pdf/operations/PDFExtractTextOperation.java

```diff
@@ -70,8 +76,13 @@
     public DocumentModel run(DocumentModel input) throws IOException {
         PDFTextExtractor textExtractor = new PDFTextExtractor(input, pdfxpath);
         textExtractor.setPassword(password);
-        String extractedText = removepatternfromresult ?
-            textExtractor.extractLastPartOfLine(patterntofind) : textExtractor.extractLineOf(patterntofind);
+        String extractedText;
+        if (StringUtils.isBlank(patterntofind)) {
+            extractedText = textExtractor.getAllExtractedLines();
+        } else {
+            extractedText = removepatternfromresult ? textExtractor.extractLastPartOfLine(patterntofind)
+                    : textExtractor.extractLineOf(patterntofind);
+        }
```

Comment:
Sorry last one. For readability could you write this as:

        if (StringUtils.isBlank(patterntofind)) {
             extractedText = textExtractor.getAllExtractedLines();
         } else if (removepatternfromresult) {
             extractedText = textExtractor.extractLastPartOfLine(patterntofind);
         } else {
             extractedText = textExtractor.extractLineOf(patterntofind);
         }

Reply:
Yes I can :-) (done)

Reply:
Thanks